File: LeetCode/0027. Remove Element/solution.py
"""
113 / 113 test cases passed.
Runtime: 40 ms
Memory Usage: 15 MB
"""
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        size = 0
        for num in nums:
            if num != val:
                nums[size] = num
                size += 1
        return size

File: LeetCode/0027. Remove Element/Solution.cpp
/**
 * 113 / 113 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 8.6 MB 
 */
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int len = 0;
        for (int i = 0; i < nums.size(); ++ i) {
            if (nums[i] != val) {
                nums[len++] = nums[i];
            }
        }
        return len;
    }
};

/**
 * 113 / 113 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 8.5 MB 
 */
class Solution2 {
public:
    int removeElement(vector<int>& nums, int val) {
        int n = nums.size() - 1;
        for (int i = 0; i <= n; ++ i) {
            if (nums[i] == val) {
                swap(nums[i--], nums[n--]);
            }
        }
        return n + 1;
    }
};
